### PR TITLE
Created axiosInstance for making API requests to the backend

### PR DESCRIPTION
**Pull request by Punithshetty321**

### CODE REVIEW

File: client/configs/axios.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import axios from 'axios'
+
+export const axiosInstance = axios.create({
+    baseURL: 'http://localhost:4000'
```
Comment (pavankpdev) [inline client/configs/axios.js]: use environment variables to specify the base URL, [refer this doc](https://nextjs.org/docs/pages/building-your-application/configuring/environment-variables)

File: client/package.json [hunk 2]
```diff
@@ -10,7 +10,8 @@
   },
   "dependencies": {
     "autoprefixer": "10.4.14",
-    "axios": "^1.6.0",
+    "axios": "^1.6.2",
+    "cors": "^2.8.5",
```
Comment (pavankpdev) [inline client/package.json]: remove this package, `cors` is not required for frontend

File: server/routes/auth.js [hunk 3]
```diff
@@ -1,5 +1,5 @@
 import express from 'express';
-import { registerUser, loginUser, } from '../controller/auth.js';
+import { registerUser, loginUser, } from '../controller/Auth.js';
```
Comment (pavankpdev) [inline server/routes/auth.js:2]: Was this filename change verified? 
If not do it by running the server

**Review by pavankpdev — CHANGES_REQUESTED** [review]

File: client/configs/axios.js [hunk 4]
```diff
@@ -1,5 +1,5 @@
 import axios from 'axios'
 
 export const axiosInstance = axios.create({
-    baseURL: 'http://localhost:4000'
+    baseURL: process.env.BASE_URL,
```
Comment (pavankpdev) [inline client/configs/axios.js]: Was this tested?

File: client/package.json [hunk 5]
```diff
@@ -11,7 +11,7 @@
   "dependencies": {
     "autoprefixer": "10.4.14",
     "axios": "^1.6.2",
-    "cors": "^2.8.5",
+    "dotenv": "^16.3.1",
```
Comment (pavankpdev) [inline client/package.json:14]: dotenv is not required, NextJS as support for ENVs by default

**Review by pavankpdev — CHANGES_REQUESTED** [review]
Need the confirmation on the testing status for the new ENV change

File: client/configs/axios.js (hunk 4 above)
Comment (Punithshetty321) [reply client/configs/axios.js]: yes its working

**Review by pavankpdev — APPROVED** [review]
looks good

**Pull request merged**